server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala

```diff
@@ -180,6 +181,160 @@ class DeltaSharedTable(
     snapshot.version
   }
 
+  private def getResponseProtocol(p: Protocol, responseFormat: String): Object = {
+    if (responseFormat == "delta") {
+      model.DeltaProtocol(p.minReaderVersion).wrap
+    } else {
+      model.Protocol(p.minReaderVersion).wrap
+    }
+  }
+
+  private def getResponseMetadata(
+      m: Metadata,
+      startingVersion: Option[Long],
+      responseFormat: String
+  ): Object = {
+    if (responseFormat == "delta") {
+      model.DeltaMetadata(
+        id = m.id,
+        name = m.name,
+        description = m.description,
+        format = model.Format(),
+        schemaString = cleanUpTableSchema(m.schemaString),
+        partitionColumns = m.partitionColumns,
+        configuration = m.configuration,
+        version = if (startingVersion.isDefined) {
+          startingVersion.get
+        } else {
+          null
+        },
+        createdTime = m.createdTime
+      ).wrap
+    } else {
+      model.Metadata(
+        id = m.id,
+        name = m.name,
+        description = m.description,
+        format = model.Format(),
+        schemaString = cleanUpTableSchema(m.schemaString),
+        configuration = getMetadataConfiguration(m.configuration),
+        partitionColumns = m.partitionColumns,
+        version = if (startingVersion.isDefined) {
+          startingVersion.get
+        } else {
+          null
+        }
+      ).wrap
+    }
+  }
+
+  private def getResponseAddFile(
+      addFile: AddFile,
+      signedUrl: PreSignedUrl,
+      version: java.lang.Long,
+      timestamp: java.lang.Long,
+      responseFormat: String,
+      returnAddFileForCDF: Boolean = false): Object = {
+    if (responseFormat == "delta") {
+      model.DeltaAddFile(
+        path = signedUrl.url,
+        id = Hashing.md5().hashString(addFile.path, UTF_8).toString,
+        expirationTimestamp = signedUrl.expirationTimestamp,
+        partitionValues = addFile.partitionValues,
+        size = addFile.size,
+        modificationTime = addFile.modificationTime,
+        dataChange = addFile.dataChange,
+        stats = addFile.stats,
+        version = version,
+        timestamp = timestamp
+      ).wrap
+    } else if (returnAddFileForCDF) {
+      model.AddFileForCDF(
```

Comment:
Question: why do have AddFileForCDF which is not needed in delta? considering adding a comment.

Reply:
Because, 
1) we need to serialize these actions as json, and when we develop cdf, AddFile is used as "file", so we need another action to be "add". 
2) And "file" suppose not to contain fields such as version/timestamp.